### ISSUE TITLE
Conform to DIMACS output

### ISSUE THREAD

**Issue by weaversa** [Z3Prover/z3#5261]
Would you be willing to consider conforming to the SAT competition output when printing results from CNF input? For example, `z3` currently produces

```
$ z3 mySATfile.cnf
sat
-1 2 3 -4
```

whereas SAT solvers (at least those compatible with the competition) have output that looks like:

```
$ z3 mySATfile.cnf
s SATISFIABLE
v -1 2 3 -4
```
or
```
$ z3 myUNSATfile.cnf
s UNSATISFIABLE
```

Some documentation on the output format can be found [here](https://satcompetition.github.io/2021/track_main.html)

This would make `z3` work seamlessly as a drop-in replacement in any cli-based application that currently uses a CNF-based SAT solver. Thank you.

**Comment by NikolajBjorner** [comment]
is this better?

**Comment by weaversa** [comment]
Yes, that looks perfect for SAT. However, I'm unaware if there is a standard for cores (your `v core` change). If you're comfortable, I'm satisfied with the changes you've made. Thank you!

**Comment by lou1306** [comment]
I am sorry to dig up a year-old issue, but shouldn't the solution line (i.e., the one starting with `v`) end with a `0`?

In other words: the output from the 1st message in this issue should actually look like this:

```
$ z3 mySATfile.cnf
s SATISFIABLE
v -1 2 3 -4 0
```

I could not find an official specification, but this is how minisat, glucose, and picosat all behave.

In case, I would be available to submit a pull request (this could likely be done in one line of code, after all).

**Comment by NikolajBjorner** [comment]
can you confirm if this does the trick?

**Comment by lou1306** [comment]
Actually, you should avoid having two whitespace characters bofore the 0. I am using CBMC (https://github.com/diffblue/cbmc/) to see if it can understand z3's output, and its parser seems to reject the double whitespace. Again, sadly I cannot tell if this is just a CBMC thing or it is mandated in some DIMACS format specification.

Anyway, `std::cout << "0\n";` should do the trick.
